jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -569,6 +575,10 @@ ecma_string_to_number (const ecma_string_t *str_p) /**< ecma-string */
 
       return num;
     }
+    case ECMA_STRING_CONTAINER__COUNT:
+    {
+      JERRY_UNREACHABLE ();
+    }
```

Comment:
This is indeed a duplication and one of the asserts is not needed. The `JERRY_UNREACHABLE();` below, after the switch, acts exactly as if `default: JERRY_UNREACHABLE();` was written in the switch. So, either just drop the `case ECMA_STRING_CONTAINER__COUNT:` or drop the `JERRY_UNREACHABLE();` below (but in the second case, rename the case to `default:`. Thus, no mentioning of `ECMA_STRING_CONTAINER__COUNT` is needed in either case.)